### DISCUSSION TITLE
Postfix config for irrd_submit_email

### DISCUSSION THREAD

**Discussion by dhutchin-ctl** [irrdnet/irrd#675] [Q&A]
I looked in the documentation and in discussions and did not find a good example of configuring email into irrd_submit_email.  That seems to be left as a system admin exercise for the user.  A couple of example configurations would be nice.  I am most interested in an example with postfix.  I tried the old config for the *OLD* version of IRRd from RIPE (was that version 1?) and ran into all sorts of permissions problems.  Further reading suggests modifications to /etc/postfix/{address.txt, main.cf, master.cf}.  But instead of taking a long time through trial and error, looking for the quick-notes version of how to do this. 

I created users auto-dbm and rpsl (both with same id), maybe I did not need to do that.  But I do want to log *ALL* incoming email to a file simultaneously with sending it to irrd_submit_email.  My old .forward was:
\auto-dbm, "|/export/home/irrd/.venv/bin/irrd_submit_email --config=/app/irrd/etc/irrd.yaml"
so that I had the log of all incoming mail in /var/spool/mail/auto-dbm.

Anyway, I have tested catting a mail file into irrd_submit_email as the irrd user, and that seems to work fine.  So just looking for good examples of the postfix config either here in the discussions or in the documentation.

**Comment by rubenskuhl** [comment]
We use procmail to send submissions to the submit script:
grep procmail /etc/postfix/main.cf
mailbox_command = /usr/bin/procmail

We use the irrd user to process mail submissions:
 cat /etc/aliases
 auto-dbm: irrd
 
And use this ruleset for procmail:
 
cat /home/irrd/.procmailrc
```LOGFILE=$HOME/logs/procmail.log
#VERBOSE=yes
:0
* ! ^From:.*@bgp\.net\.br
|/home/irrd/irrd-venv/bin/irrd_submit_email

**Comment by mxsasha** [comment]
I use opensmtpd with an alias:
```
auto-dbm: |/home/irrd/irrd-venv/bin/python /home/irrd/irrd/irrd/scripts/submit_email.py
```
Slightly different paths because this runs from a git checkout for development, yours looks fine for an installed version.

One thing to keep in mind is whether the user this will run as has permissions to read the config file and access PostgreSQL. Or run through sudo. Other than that, can't say much without more logs.
